File: questions/28_index_of_first_occurrence/Javascript/code2.js
function strStr(haystack, needle) {
    if (needle === "") return -1; // empty needle

    let n = haystack.length, m = needle.length;

    for (let i = 0; i <= n - m; i++) { 
        if (haystack.substring(i, i + m) === needle) { 
            return i;
        }
    }
    return -1; 
}

File: questions/28_index_of_first_occurrence/CPP/code2.cpp
#include <string>
using namespace std;

int strStr(string haystack, string needle) {
    if (needle.empty()) return 0; // empty needle

    int n = haystack.size(), m = needle.size();

    for (int i = 0; i <= n - m; i++) { 
        if (haystack.substr(i, m) == needle) { 
            return i;
        }
    }
    return -1; 
}
